tests/nn/pipe_process/test_pipe.py

```diff
@@ -635,7 +635,7 @@ def deferred_batch_norm_params(checkpoint, lazy):
     assert torch.allclose(pipe[0].bias.grad, bn.bias.grad, atol=1e-4)
 
 
-@torch_spawn([5])
+@torch_spawn([4])
```

Comment:
why did you need this? was it causing an error? you should also change line 649 to be `if model.group.rank() == 3:`

Reply:
NCCL 2.7.8 does not support two processes having a context to the same GPU. Our circleci GPU instance only has 4 GPUs.